server/src/main/scala/ch.epfl.scala.index.server/BadgesSupport.scala

```diff
@@ -10,93 +10,203 @@ object BadgesSupport {
+        )(SemanticVersion.ordering.reverse)
+    } yield s"$artifactVersion (${summarise(scalaTargetsByArtifactVersion(artifactVersion), keyVersions)})")
+      .mkString(", ")
+
+    def summarise(scalaTargets: Set[T], interestingKeyVersions: Set[V]): String
+  }
 
-    val platformEditionsSupportedForAllLanguageVersions: Set[PlatformEdition] =
-      scalaTargetsByLanguageVersion.values
-        .map(_.collect { case st: ScalaTargetWithPlatformBinaryVersion =>
-          st.platformEdition
-        })
-        .reduce(_ & _)
-
-    val platformBinaryVersionsByTargetType
-        : SortedMap[ScalaTargetType, SortedSet[BinaryVersion]] =
-      SortedMap.from(
-        platformEditionsSupportedForAllLanguageVersions
-          .groupMap(_.targetType)(_.version)
+  /**
+   * Summarises primarily the available Scala LanguageVersions - but
+   * will secondarily summarise the PlatformEditions that are supported
+   * for all those Scala LanguageVersions.
+   */
+  object SummariseLanguageVersions
+      extends SummaryStrategy[ScalaTarget, LanguageVersion] {
+    override def versionFor(t: ScalaTarget): LanguageVersion = t.languageVersion
+    override def removeSuperfluousVersionsFrom(
+        versions: Set[LanguageVersion]
+    ): Set[
+      LanguageVersion
+    ] = // we're only interested in the latest version of a binary family
+      versions.groupBy(_.family).values.map(_.max).toSet
+
+    override def summarise(
+        scalaTargets: Set[ScalaTarget],
+        interestingKeyVersions: Set[LanguageVersion]
+    ): String = {
+      val scalaTargetsByLanguageVersion
+          : Map[LanguageVersion, Set[ScalaTarget]] =
+        scalaTargets
+          .groupBy(_.languageVersion)
           .view
-          .mapValues(SortedSet.from(_)(BinaryVersion.ordering.reverse))
-      )
+          .filterKeys(interestingKeyVersions)
+          .toMap
 
-    val targetsSummary: Option[String] =
-      Option.when(platformBinaryVersionsByTargetType.nonEmpty)(
-        " - " + platformBinaryVersionsByTargetType
-          .map { case (targetType, platformBinaryVersions) =>
-            s"$targetType ${platformBinaryVersions.mkString("+")}"
-          }
-          .mkString(", ")
-      )
+      val languageVersions: SortedSet[LanguageVersion] =
+        SortedSet.from(scalaTargetsByLanguageVersion.keySet)(
+          LanguageVersion.ordering.reverse
+        )
+
+      val platformEditionsSupportedForAllLanguageVersions
+          : Set[PlatformEdition] =
+        scalaTargetsByLanguageVersion.values
+          .map(_.collect { case st: ScalaTargetWithPlatformBinaryVersion =>
+            st.platformEdition
+          })
+          .reduce(_ & _)
+
+      val targetsSummary: Option[String] =
+        Option.when(platformEditionsSupportedForAllLanguageVersions.nonEmpty)(
+          " - " + summarisePlatformTargets(
+            platformEditionsSupportedForAllLanguageVersions,
+            "+"
```

Comment:
Why using a different character in platforms compare to Scala versions?
I was surprised by that when looking at this badge:
![image](https://user-images.githubusercontent.com/13123162/124445688-0dae2100-dd80-11eb-94b5-1845f581a344.png)

Reply:
The `+` is current behaviour for the badge - but you don't actually get to see it...! I was thinking of an older version of the badge (that no longer exists!) that would show multiple platform types in the same badge, eg

`cats-core 2.6.1 (Scala 3, 2.13, 2.12 - JS 1.x+0.6, Native 0.4+0.3)`

I thought if the platform editions were separated by `,` rather than `+` that would become less clear - it would look like:

`cats-core 2.6.1 (Scala 3, 2.13, 2.12 - JS 1.x, 0.6, Native 0.4, 0.3)`

'Native' there seems to be mixed in with some random version numbers, so I thought `+` would at least group the platform versions by platform type.

However, thanks to [feedback](https://github.com/scalacenter/scaladex/issues/659#issuecomment-826381083) the badge now only ever shows one platform at most, so the situation only ever gets this bad:

`cats-core 2.6.1 (Scala 3, 2.13, 2.12 - JS 1.x, 0.6)`

...that does look ok to me, so I'l take out the custom separator parameter!

Reply:
That's fixed with 34949f3!